app.js

```diff
@@ -1499,6 +1499,180 @@ app.delete("/room/:roomId", function (req, res) {
     });
 });
 
+
+//suggest meeting times given list of participants
+app.post("/api/meetingSuggestion", function (req, res) {
+
+    //currently only look 3 days ahead, and assume 7AM - 5PM workday. only search on weekdays
+
+    if(!req.body.participants) {
+      res.statusCode = 400;
+      console.log();
+      res.json({
+        "requestURL":  "/room",
+        "action": "post",
+        "status": 400,
+        "message": "Bad Request",
+        "timestamp": new Date()
+      });
+    }
+
+    else {
+
+      var participants = req.body.participants || null;
+
+      //using ADDDATE(CURDATE(), 4) just to limit the amount of meetings to search through later
+      var getMeetingSql = "SELECT DISTINCT(start_datetime), end_datetime FROM ebdb.meeting WHERE start_datetime >= CURDATE() AND end_datetime < ADDDATE(CURDATE(), 4) AND "
```

Comment:
Added filters for meetings before/after specified time. Also added DISTINCT(start_datetime) to filter out duplicate meetings.

Reply:
👍